singularity-eos/eos/eos_powermg.hpp

```diff
@@ -0,0 +1,441 @@
+//------------------------------------------------------------------------------
+// © 2021-2024. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#ifndef _SINGULARITY_EOS_EOS_POWERMG_HPP_
+#define _SINGULARITY_EOS_EOS_POWERMG_HPP_
+
+#include <cmath>
+#include <cstdio>
+
+#include <limits>
+
+#include <ports-of-call/portable_errors.hpp>
+#include <singularity-eos/base/constants.hpp>
+#include <singularity-eos/base/math_utils.hpp>
+#include <singularity-eos/base/robust_utils.hpp>
+#include <singularity-eos/eos/eos_base.hpp>
+
+namespace singularity {
+
+using namespace eos_base;
+using namespace robust;
+
+// COMMENT: This is a complete, thermodynamically consistent Mie-Gruneisen
+// EOS based on a linear Us-up Hugoniot, Us=Cs+s*up.
```

Comment:
This EOS is a power-law series Us-up Hugoniot, right?

Reply:
Whoops

Reply:
Done